example_deploy.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import re
+import shlex
+import subprocess
+import sys
+from pathlib import Path
+from typing import NamedTuple, Optional
+
+from example_utils import get_examples, ExampleType
+
+
+class DeployError(NamedTuple):
+    stdout: str
+    stderr: str
+    code: int
+
+
+def deploy(
+    deployable: bool, module_with_stub: Path, dry_run: bool, filter_pttrn: Optional[str]
+) -> Optional[DeployError]:
+    if filter_pttrn and not re.match(filter_pttrn, module_with_stub.name):
+        return None
+
+    if not deployable:
+        print(f"⏩ skipping: '{module_with_stub.name}' is not marked for deploy")
+        return None
+
+    deploy_command = f"modal app deploy {module_with_stub.name}"
+    if dry_run:
+        print(f"🌵  dry-run: '{module_with_stub.name}' would have deployed")
+    else:
+        print(f"⛴ deploying: '{module_with_stub.name}' ...")
+        r = subprocess.run(shlex.split(deploy_command), cwd=module_with_stub.parent, capture_output=True)
+        if r.returncode != 0:
+            print(f"⚠️ deployment failed: '{module_with_stub.name}'", file=sys.stderr)
+            return DeployError(stdout=r.stdout, stderr=r.stderr, code=r.returncode)
+        else:
+            print(f"✔️ deployed '{module_with_stub.name}")
+    return None
+
+
+def main(argv: Optional[list[str]] = None) -> int:
+    parser = argparse.ArgumentParser(
+        description="Deploy Modal example programs to our Modal organization.",
+        add_help=True,
+    )
+    parser.add_argument(
+        "--dry-run", default=True, action="store_true", help="show what apps be deployed without deploying them."
+    )
+    parser.add_argument("--no-dry-run", dest="dry_run", action="store_false")
+    parser.add_argument(
+        "--filter",
+        default=None,
+        help="Filter which apps are deployed with basic pattern matching. eg. 'cron' matches 'say_hello_cron.py'.",
+    )
+    arguments = parser.parse_args()
+
+    if arguments.dry_run:
+        print("INFO: dry-run is active. Intended deployments will be displayed to console.")
+
+    example_modules = (ex for ex in get_examples() if ex.type == ExampleType.MODULE)
+    filter_pttrn = (r".*" + arguments.filter + r".*") if arguments.filter else None
+    results = [
+        deploy(
+            deployable=bool(ex_mod.metadata.get("deploy")),
```

Comment:
we should probably just parse the metadata directly in `example_utils` and write it as attributes to the dataclass, but that's not necessarily within scope for this PR